micro_manager/micro_manager.py

```diff
@@ -437,13 +438,45 @@ def _solve_micro_simulations(self, micro_sims_input: list) -> list:
         micro_sims_output = [None] * self._local_number_of_sims
 
         for count, sim in enumerate(self._micro_sims):
-            start_time = time.time()
-            micro_sims_output[count] = sim.solve(micro_sims_input[count], self._dt)
-            end_time = time.time()
 
-            if self._is_micro_solve_time_required:
+            if not self._crashed_sims[count]:
+                try:
+                    start_time = time.time()
+                    micro_sims_output[count] = sim.solve(
+                        micro_sims_input[count], self._dt)
+                    end_time = time.time()
+                except Exception as error_message:
+                    _, mesh_vertex_coords = self._participant.get_mesh_vertex_ids_and_coordinates(
+                        self._macro_mesh_name)
```

Comment:
Rather than having the API call here, we can just save the coordinates and sample them as and when necessary.